consensus/common_test.go

```diff
@@ -37,6 +37,23 @@ func ensureDir(dir string, mode os.FileMode) {
 	}
 }
 
+type NodeConfig struct {
+	cfg.Config `mapstructure:",squash"`
+	P2P        *p2p.Config   `mapstructure:"p2p"`
+	Mempool    *mempl.Config `mapstructure:"mempool"`
+	Consensus  *Config       `mapstructure:"consensus"`
+}
+
+// TODO: This is the same as NodeConfig. Should we move the configs to types (?)
```

Comment:
Hmm. `p2p` doesn't currently import `types` and I don't think it should. and `types` shouldn't import `p2p`. So unless we remove p2p options from the config file, we're back to square one. 

Maybe we should be thinking about why some packages need to read the whole config file. This all arose because `consensus` ends up using the same full config as `node`, since it uses p2p and mempool, but has to duplicate the definition to avoid import cycle. What if instead `consensus` only loads its own config, and uses defaults for mempool and p2p. Then we can mostly keep things as they are, and not have the duplicated definition, but we would lose the ability to change p2p parameters from the config file in the consensus unit tests. That said, we'd still be doing this once in each package for tests:

```
func ResetConfig(name string) *Config {
        viperConfig := tendermint_test.ResetConfig(name)
        config := new(struct {
                cfg.Config `mapstructure:",squash"`
                Mempool    *Config `mapstructure:"mempool"`
        })  
        if err := viperConfig.Unmarshal(config); err != nil {
                panic(err)
        }   
        return config.Mempool

}
```